### PR TITLE
Add AlwaysParentSample sampler

### PR DESCRIPTION
**Pull request by amrmahdi**
Per the `ALWAYS_PARENT` sampler from the spec   https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/sdk-tracing.md#built-in-samplers

### CODE REVIEW

File: sdk/trace/sampling.go [hunk 1]
```diff
@@ -77,3 +77,10 @@ func NeverSample() Sampler {
 		return SamplingDecision{Sample: false}
 	}
 }
+
+// AlwaysParentSample returns a Sampler that samples only if the parent span is sampled.
```
Comment (krnowak) [inline sdk/trace/sampling.go]: How about:

```suggestion
// AlwaysParentSample returns a Sampler that samples a trace only
// if the parent span is sampled.
```

so it's more in line with docs for other samplers, and have a wrapped line.

File: sdk/trace/sampling_test.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
+// Copyright 2019, OpenTelemetry Authors
```
Comment (rghetia) [inline sdk/trace/sampling_test.go]: s/2019/2020/

**Comment by amrmahdi** [comment]
> What should be the behavior if this is a root span? It is more of an issue in the spec since it doesn't define this. If the desired behavior for root is
> 
> * always sample then it is same as probabilitySampler with probability of 1.
> * if do not sample then it is same as probabilitySampler with probability of 0.
> * if sample as certain probability x then it is same as probabilitySampler with x.
> 
> Essentially it is same as probabilitySampler.
> 
> The other thing is that AlwaysParentSampler is same as ProbabilitySampler with probability of 1. If there is a parent and then simply follow the decision on the parent. That is same as AlwaysParentSample (if we clarify what should be the behavior for root span for AlwaysParentSampler).

This is a good point. I believe it should be equivalent to ProbabilitySampler with 0. And then we can just wrap the ProbabilitySampler with 0 inside the AlwaysParentSampler and clarify it in the docs.

**Comment by rghetia** [comment]
> This is a good point. I believe it should be equivalent to ProbabilitySampler with 0. And then we can just wrap the ProbabilitySampler with 0 inside the AlwaysParentSampler and clarify it in the docs.

I agree on ProbabilitySampler with 0.
Actually, spec implicitly implies that on root span do not sample. From spec `Returns RECORD_AND_SAMPLED if SampledFlag is set to true on parent SpanContext and NOT_RECORD otherwise.`

**Pull request merged**